Cannot run installations of the JVM

<!--

    The template below contains optional suggestions. You
    can omit some information.

-->

## The Problem
I can't run this with an installation of Rakudo on the JVM in my `$PATH`:
```
$ rakudo-j -e 'say 1'
```

## Expected Behavior
This should output `1`.

## Actual Behavior
An error gets thrown:
```
bastille% rakudo-j -e 'say 1'
Error: Could not find or load main class perl6
Caused by: java.lang.ClassNotFoundException: perl6
```

## Steps to Reproduce
<!-- Provide a set of steps to reproduce the problem.
     If the program unexpectedly crashes, please run it with
     `perl6 --ll-exception` or  `raku --ll-exception` and provide the produced output. -->
Install Rakudo on the JVM and run the test command.

## Environment
* Operating system:
OpenBSD bastille.kennel.qt 6.7 GENERIC.MP#7 amd64
* Compiler version (`perl6 -v` or `raku -v`):
Can't tell because the same error gets thrown. This is a build of Rakudo on the JVM on a branch forked some point after v2020.07 was released.

--- a/tools/build/create-jvm-runner.pl
+++ b/tools/build/create-jvm-runner.pl
@@ -99,6 +99,7 @@ elsif ($type eq 'install') {
             $preamble_unix,
             ": \${NQP_HOME:=\"\$DIR/../share/nqp\"}",
             ": \${NQP_JARS:=\"$nqpjars\"}",
+            ": \${RAKUDO_HOME:=\"\$PERL6_HOME\"}",
             ": \${RAKUDO_HOME:=\"\$DIR/../share/perl6\"}",
             ": \${RAKUDO_JARS:=\"$rakudo_jars\"}",
             "exec "
@@ -109,6 +110,7 @@ elsif ($type eq 'install') {
             $preamble_unix,
             ": \${NQP_HOME:=\"$static_nqp_home\"}",
             ": \${NQP_JARS:=\"$nqpjars\"}",
+            ": \${RAKUDO_HOME:=\"\$PERL6_HOME\"}",
             ": \${RAKUDO_HOME:=\"$static_rakudo_home\"}",
             ": \${RAKUDO_JARS:=\"$rakudo_jars\"}",
             "exec "

--- a/tools/build/create-jvm-runner.pl
+++ b/tools/build/create-jvm-runner.pl
@@ -28,12 +28,15 @@ my $bat       = $^O eq 'MSWin32' ? '.bat' : '';
 my $nqplibdir = $^O eq 'MSWin32' ? File::Spec->catfile($static_nqp_home, 'lib') : File::Spec->catfile('${NQP_HOME}', 'lib');
 
 my $nqpjars;
-if ($^O eq 'MSWin32' || !$relocatable) {
+if ($^O eq 'MSWin32') {
     $nqpjars = $thirdpartyjars;
 }
-elsif ($relocatable) {
+else {
+    # The following is a workaround turning the third-party JARS into relative
+    # paths. The clean solution would be to pass in relative paths and only
+    # prefix those with ${NQP_HOME} here.
     my @thirdpartyjars = map { abs_path($_) } split($cpsep, $thirdpartyjars);
-    my $nqp_home       = abs_path(File::Spec->catpath($prefix, 'share', 'nqp'));
+    my $nqp_home       = File::Spec->catdir(abs_path($prefix), 'share', 'nqp');
     @thirdpartyjars    = map { $_ =~ s,$nqp_home,\${NQP_HOME},; $_ } @thirdpartyjars;
     $nqpjars           = join($cpsep, @thirdpartyjars);
 }

